aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -326,6 +344,10 @@ SparseTensor& add_out_sparse_cpu(SparseTensor& r, const SparseTensor& t, const S
         }
     );
 
+    if (r.scalar_type() != commonDtype) {
+      r_values = r_values.to(r.scalar_type());
```

Comment:
Instead of doing a `.to`, we can save memory here by doing a copy from `r_values` directly into the values tensor instead of creating a new tensor here and assigning it.

Reply:
I guess the values tensor of the result might not be large enough to hold all of that...

Reply:
I think for this one we may still need the `to`, otherwise we would have to create a new empty tensor first to copy values into anyway, and I assume that's just basically the implementation of `to`? 

`r_values` here is the promoted version (e.g. float += double, r_values is a double tensor, but we'd need to set r._values to a yet-uncreated float tensor containing those values.

Reply:
`r` is the result tensor. It's possible that `r` already has data and `r._values()` and `r._indices()` are tensors with some non-trivial storage. If that is the case, instead of discarding values and indices in favor of the new values, we could copy r_values directly into `r._values()`, assuming there is enough memory in `r._values()`.